pkg/nmi/server/linuxServer.go

```diff
@@ -0,0 +1,90 @@
+// +build linuxos
+
+package server
+
+import (
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	iptables "github.com/Azure/aad-pod-identity/pkg/nmi/iptables"
+	"k8s.io/klog"
+)
+
+type LinuxServer struct {
+	Server *Server
+}
+
+func RunServer(s *Server) {
```

Comment:
>RunServer [](start = 5, length = 9)

Why do you have to duplicate "Run" logic for both Windows and Linux.
I would simply keep original server.Run() implementation and add one parameter to the Run() function

Reply:
Good points. I will check the way you suggested.

Reply:
I create a interface RedirectorInt in this file and use two new files redirector_windows.go and rifirector_linux.go to impleemnt the function in this interface. In this way, we do not need the two servers and we do not have RunServer() defined twice.